pallets/committee/src/lib.rs

```diff
@@ -522,7 +538,13 @@ pub mod pallet {
 				Error::<T>::InvalidVotingPeriod
 			);
 
-			VotingPeriod::<T>::set(voting_period);
+			PendingVotingPeriod::<T>::set(Some(voting_period));
+			Self::deposit_event(Event::NewPendingVotingPeriod(
+				voting_period,
+				Self::get_next_voting_period_end(&frame_system::Pallet::<T>::block_number())?
+					.saturating_sub(T::ProposalSubmissionPeriod::get())
+					.saturating_sub(VotingPeriod::<T>::get()),
+			));
```

Comment:
Modified this to the `next_voting_circle` which matches the check-and-update of upkeep